src/main/java/com/squareup/catalog/demo/example/RetrieveCatalogObjectExample.java

```diff
@@ -0,0 +1,247 @@
+ * limitations under the License.
+ */
+package com.squareup.catalog.demo.example;
+
+import com.squareup.catalog.demo.Logger;
+import com.squareup.catalog.demo.util.Moneys;
+import com.squareup.connect.ApiException;
+import com.squareup.connect.api.CatalogApi;
+import com.squareup.connect.api.LocationsApi;
+import com.squareup.connect.models.CatalogCategory;
+import com.squareup.connect.models.CatalogDiscount;
+import com.squareup.connect.models.CatalogItem;
+import com.squareup.connect.models.CatalogItemModifierListInfo;
+import com.squareup.connect.models.CatalogItemVariation;
+import com.squareup.connect.models.CatalogModifier;
+import com.squareup.connect.models.CatalogModifierList;
+import com.squareup.connect.models.CatalogObject;
+import com.squareup.connect.models.CatalogTax;
+import com.squareup.connect.models.RetrieveCatalogObjectResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This example retrieves a single catalog object by ID and all related objects.
+ */
+public class RetrieveCatalogObjectExample extends Example {
+
+  public RetrieveCatalogObjectExample(Logger logger) {
+    super("retrieve_catalog_object", "Retrieve a catalog object by ID.", logger);
+  }
+
+  @Override
+  public void execute(CatalogApi catalogApi, LocationsApi locationsApi) throws ApiException {
+    String catalogObjectId = promptUserInput("Enter catalog object ID: ");
+
+    // Send a request to retrieve the catalog object by ID. The second boolean arguments indicates
+    // the we want related objects, such as the taxes linked to an item.
+    RetrieveCatalogObjectResponse response =
+        catalogApi.retrieveCatalogObject(catalogObjectId, true);
+    if (checkAndLogErrors(response.getErrors())) {
+      return;
+    }
+
+    // Put the related objects into a map keyed by catalog object ID so we can look them up faster.
+    HashMap<String, CatalogObject> relatedObjectsMap = new HashMap<>();
+    for (CatalogObject relatedObject : response.getRelatedObjects()) {
+      String relatedObjectId = relatedObject.getId();
+      relatedObjectsMap.put(relatedObjectId, relatedObject);
+    }
+
+    CatalogObject catalogObject = response.getObject();
+    logCatalogObjectDetails(catalogObject, relatedObjectsMap);
+  }
+
+  /**
+   * Log info about the retrieved catalog object based on it's type.
+   *
+   * @param catalogObject the {@link CatalogObject} to log info about
+   * @param relatedObjectsMap a map of related objects keyed by their IDs
+   */
+  void logCatalogObjectDetails(CatalogObject catalogObject,
+      Map<String, CatalogObject> relatedObjectsMap) {
+    CatalogObject.TypeEnum type = catalogObject.getType();
+    switch (type) {
+      case CATEGORY:
+        logCategoryDetails(catalogObject);
+        break;
+      case DISCOUNT:
+        logDiscountDetails(catalogObject);
+        break;
+      case ITEM:
+        logItemDetails(catalogObject, relatedObjectsMap);
```

Comment:
So ITEM is the only type with related objects? Would we want to log the item when we're logging ITEM_VARIATION below?